src/Hint/Lambda.hs

```diff
@@ -149,11 +151,13 @@ lambdaDecl
 
           (finalpats, body) = fromLambda . lambda pats $ origBody
           (pats2, bod2) = etaReduce pats origBody
-          template fps = unsafePrettyPrint $ reform (zipWith munge ['a'..'z'] fps) varBody
-          subts fps b = ("body", toSS b) : zipWith (\x y -> ([x],y)) ['a'..'z'] (map toSS fps)
+          template fps =
+            let (origPats, subtsVars) = mkOrigPats fps
+             in (unsafePrettyPrint (reform origPats varBody), subtsVars)
+          subts fps b = ("body", toSS b) : zipWith (\x y -> ([x],y)) subtsVars (map toSS fps)
           s1 = subts finalpats body
           --s2 = subts pats2 bod2
```

Comment:
Delete these comments if they don't mean anything (I imagine they were from when we had both HSE and GHC?)

Reply:
They are supposed to be un-commented when eta-reduce refactoring is supported. But there's some challenge in refactoring eta-reduce and it's probably not happening soon, so I'll delete them.